# Daffa Asyqar Ahmad Khalisheka

# 1103200034

# Machine Learning Class

# Telkom University

In [49]:
# Import library Google Colab untuk mengakses Google Drive
from google.colab import drive

# Mount (pasang) Google Drive pada direktori /content/drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Kode di atas digunakan dalam lingkungan Google Colab untuk mengakses Google Drive. Berikut penjelasan dari setiap baris kode:

1. `from google.colab import drive`: Ini adalah pernyataan impor yang mengimpor modul `drive` dari pustaka `google.colab`. Modul ini digunakan untuk mengakses Google Drive dari lingkungan Google Colab.

2. `drive.mount('/content/drive')`: Pada baris ini, kita memanggil metode `mount` dari modul `drive` yang telah diimpor sebelumnya. Metode ini digunakan untuk "mount" atau "pasang" Google Drive ke dalam direktori tertentu dalam lingkungan Google Colab. Di sini, Google Drive akan di-"mount" pada direktori `/content/drive`.

Dengan kata lain, kode ini menghubungkan Google Drive pengguna dengan lingkungan Google Colab, sehingga pengguna dapat mengakses dan mengelola file dan folder di Google Drive mereka langsung dari lingkungan Colab tersebut. Setelah menjalankan kode ini, Anda akan diminta untuk memberikan izin dan mengautentikasi akun Google Anda agar dapat mengakses Google Drive dari Colab.

In [50]:
# Import library Pandas dan menggantikan 'pd' sebagai alias untuk penggunaan lebih lanjut
import pandas as pd

Kode di atas adalah contoh penggunaan perintah Python yang mengimpor perpustakaan Pandas ke dalam program. Pandas adalah perpustakaan yang digunakan untuk melakukan analisis dan manipulasi data dalam Python. Pada kode tersebut, perintah `import pandas as pd` digunakan untuk mengimpor perpustakaan Pandas dan memberikan alias "pd" ke perpustakaan tersebut. Alias ini digunakan untuk memudahkan penggunaan Pandas di seluruh program, karena dengan menggunakan "pd," kita dapat mengakses berbagai fungsi dan objek Pandas tanpa harus mengetik "pandas" secara penuh setiap kali.

Dengan kata lain, kode di atas mengimpor Pandas dan menggantikan "pandas" dengan "pd" sehingga kita dapat mengakses fungsi-fungsi Pandas dengan lebih mudah, misalnya dengan menuliskan `pd.read_csv()` untuk membaca file CSV atau `pd.DataFrame()` untuk membuat kerangka data (dataframe) Pandas.

In [51]:
#Membaca file CSV yang berisi data ke dalam sebuah DataFrame
data = pd.read_csv('/content/drive/MyDrive/Dataset/melb_data.csv')

Kode di atas digunakan untuk membaca file CSV yang berisi data ke dalam sebuah DataFrame menggunakan library pandas (yang diasumsikan sudah diimpor sebelumnya).

- `data` adalah nama variabel yang digunakan untuk menyimpan DataFrame yang akan dibaca dari file CSV.
- `pd.read_csv()` adalah fungsi dari library pandas yang digunakan untuk membaca file CSV. Fungsi ini mengambil satu argumen yaitu lokasi/path dari file CSV yang akan dibaca.
- `'/content/drive/MyDrive/Dataset/melb_data.csv'` adalah argumen yang diberikan kepada fungsi `pd.read_csv()`, yang merupakan lokasi file CSV yang akan dibaca. Lokasi ini mungkin berbeda tergantung pada sistem penyimpanan file yang digunakan, dalam kasus ini asumsi bahwa file berada di direktori "/content/drive/MyDrive/Dataset/melb_data.csv".

Jadi, kode ini akan membaca data dari file CSV yang disebut "melb_data.csv" dan menyimpannya dalam bentuk DataFrame dengan nama "data" sehingga data tersebut dapat diakses dan diolah lebih lanjut menggunakan pandas.

In [52]:
#Menampilkan isi dari dataset
data

Suburb           Address  Rooms Type      Price Method  \
0         Abbotsford      85 Turner St      2    h  1480000.0      S   
1         Abbotsford   25 Bloomburg St      2    h  1035000.0      S   
2         Abbotsford      5 Charles St      3    h  1465000.0     SP   
3         Abbotsford  40 Federation La      3    h   850000.0     PI   
4         Abbotsford       55a Park St      4    h  1600000.0     VB   
...              ...               ...    ...  ...        ...    ...   
13575  Wheelers Hill      12 Strada Cr      4    h  1245000.0      S   
13576   Williamstown     77 Merrett Dr      3    h  1031000.0     SP   
13577   Williamstown       83 Power St      3    h  1170000.0      S   
13578   Williamstown      96 Verdon St      4    h  2500000.0     PI   
13579     Yarraville        6 Agnes St      4    h  1285000.0     SP   

        SellerG        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
0        Biggin   3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0   
1        Biggin   4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0   
2        Biggin   4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0   
3        Biggin   4/03/2017       2.5    3067.0  ...       2.0  1.0      94.0   
4        Nelson   4/06/2016       2.5    3067.0  ...       1.0  2.0     120.0   
...         ...         ...       ...       ...  ...       ...  ...       ...   
13575     Barry  26/08/2017      16.7    3150.0  ...       2.0  2.0     652.0   
13576  Williams  26/08/2017       6.8    3016.0  ...       2.0  2.0     333.0   
13577     Raine  26/08/2017       6.8    3016.0  ...       2.0  4.0     436.0   
13578   Sweeney  26/08/2017       6.8    3016.0  ...       1.0  5.0     866.0   
13579   Village  26/08/2017       6.3    3013.0  ...       1.0  1.0     362.0   

       BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  \
0               NaN        NaN        Yarra -37.79960   144.99840   
1              79.0     1900.0        Yarra -37.80790   144.99340   
2             150.0     1900.0        Yarra -37.80930   144.99440   
3               NaN        NaN        Yarra -37.79690   144.99690   
4             142.0     2014.0        Yarra -37.80720   144.99410   
...             ...        ...          ...       ...         ...   
13575           NaN     1981.0          NaN -37.90562   145.16761   
13576         133.0     1995.0          NaN -37.85927   144.87904   
13577           NaN     1997.0          NaN -37.85274   144.88738   
13578         157.0     1920.0          NaN -37.85908   144.89299   
13579         112.0     1920.0          NaN -37.81188   144.88449   

                       Regionname Propertycount  
0           Northern Metropolitan        4019.0  
1           Northern Metropolitan        4019.0  
2           Northern Metropolitan        4019.0  
3           Northern Metropolitan        4019.0  
4           Northern Metropolitan        4019.0  
...                           ...           ...  
13575  South-Eastern Metropolitan        7392.0  
13576        Western Metropolitan        6380.0  
13577        Western Metropolitan        6380.0  
13578        Western Metropolitan        6380.0  
13579        Western Metropolitan        6543.0  

[13580 rows x 21 columns]

Kode di atas merupakan contoh tampilan data dalam bentuk tabel atau dataframe. Data ini mungkin merupakan dataset yang digunakan dalam analisis data atau tugas pemrograman.

Dataset ini memiliki berbagai kolom dengan berbagai informasi seperti:
- Suburb: Nama daerah atau kota
- Address: Alamat properti
- Rooms: Jumlah kamar di dalam properti
- Type: Jenis properti (misalnya, 'h' mungkin mengacu pada rumah)
- Price: Harga properti
- Method: Metode penjualan properti (misalnya, 'S' untuk penjualan langsung atau 'SP' untuk penjualan pribadi)
- SellerG: Nama agen properti
- Date: Tanggal penjualan properti
- Distance: Jarak ke pusat kota
- Postcode: Kode pos properti
- ... (Ada kolom-kolom lainnya yang tidak ditampilkan dalam potongan kode ini)

Data ini dapat digunakan untuk berbagai analisis, seperti memprediksi harga properti berdasarkan fitur-fiturnya atau menjalankan statistik deskriptif untuk memahami karakteristik dataset ini.

In [53]:
# Ini adalah kode untuk menampilkan informasi tentang objek 'data'
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13580 entries, 0 to 13579
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         13580 non-null  object 
 1   Address        13580 non-null  object 
 2   Rooms          13580 non-null  int64  
 3   Type           13580 non-null  object 
 4   Price          13580 non-null  float64
 5   Method         13580 non-null  object 
 6   SellerG        13580 non-null  object 
 7   Date           13580 non-null  object 
 8   Distance       13580 non-null  float64
 9   Postcode       13580 non-null  float64
 10  Bedroom2       13580 non-null  float64
 11  Bathroom       13580 non-null  float64
 12  Car            13518 non-null  float64
 13  Landsize       13580 non-null  float64
 14  BuildingArea   7130 non-null   float64
 15  YearBuilt      8205 non-null   float64
 16  CouncilArea    12211 non-null  object 
 17  Lattitude      13580 non-null  float64
 18  Longti

Kode di atas adalah potongan dari Python yang menggunakan pustaka Pandas untuk menampilkan informasi tentang sebuah DataFrame yang disimpan dalam variabel 'data'. DataFrame ini tampaknya berisi data tentang properti-properti, mungkin dalam konteks penjualan properti. Berikut adalah penjelasan dari hasil keluaran yang ditampilkan oleh kode tersebut:

- `data.info()`: Ini adalah panggilan fungsi yang digunakan untuk menampilkan informasi tentang DataFrame 'data'. Fungsi ini akan memberikan informasi tentang jumlah baris, nama kolom, jumlah nilai non-null, tipe data kolom, dan penggunaan memori yang terkait dengan DataFrame ini.

Hasil keluaran dari kode tersebut menjelaskan hal-hal berikut:

1. `RangeIndex: 13580 entries, 0 to 13579`: Ini mengindikasikan bahwa DataFrame memiliki 13,580 baris (entri) dengan indeks (index) yang berkisar dari 0 hingga 13,579.

2. `Data columns (total 21 columns)`: Menunjukkan bahwa DataFrame ini memiliki 21 kolom.

3. Daftar kolom:
   - 'Suburb', 'Address', 'Type', 'Method', 'SellerG', 'Date', 'CouncilArea', dan 'Regionname' adalah kolom dengan tipe data objek (teks atau string).
   - 'Rooms', 'Postcode', 'Bedroom2', 'Bathroom', 'Car', 'Landsize', 'BuildingArea', 'YearBuilt', 'Lattitude', 'Longtitude', dan 'Propertycount' adalah kolom dengan tipe data numerik (int atau float).
   - 'Price' adalah kolom dengan tipe data float.
   
4. 'Non-Null Count': Menunjukkan jumlah nilai non-null (tidak ada yang hilang) dalam setiap kolom. Kolom 'Car', 'BuildingArea', 'YearBuilt', dan 'CouncilArea' memiliki beberapa nilai yang hilang (NaN).

5. 'Dtypes': Menunjukkan tipe data yang digunakan untuk setiap kolom dalam DataFrame.

6. 'Memory usage': Menunjukkan estimasi penggunaan memori untuk DataFrame ini, yaitu sekitar 2.2+ MB.

Kode ini memberikan informasi dasar tentang struktur dan tipe data dalam DataFrame 'data', yang dapat membantu dalam analisis dan pemrosesan lebih lanjut terhadap data properti ini.

In [55]:
# Menghitung jumlah nilai NaN (Not a Number) pada kolom 'Bedroom2'
nan_count_bedroom2 = data['Bedroom2'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom 'Bathroom'
nan_count_bathroom = data['Bathroom'].isna().sum()

# Menghitung jumlah nilai NaN pada kolom 'Car'
nan_count_car = data['Car'].isna().sum()

# Menampilkan jumlah NaN pada kolom 'Bedroom2' ke layar
print(f"Jumlah NaN pada kolom Bedroom2: {nan_count_bedroom2}")

# Menampilkan jumlah NaN pada kolom 'Bathroom' ke layar
print(f"Jumlah NaN pada kolom Bathroom: {nan_count_bathroom}")

# Menampilkan jumlah NaN pada kolom 'Car' ke layar
print(f"Jumlah NaN pada kolom Car: {nan_count_car}")


Jumlah NaN pada kolom Bedroom2: 0
Jumlah NaN pada kolom Bathroom: 0
Jumlah NaN pada kolom Car: 62


Kode di atas digunakan untuk menghitung jumlah nilai NaN (Not a Number) pada tiga kolom berbeda ('Bedroom2', 'Bathroom', dan 'Car') dalam sebuah dataset. Penghitungan dilakukan dengan menggunakan metode `.isna()` yang mengidentifikasi apakah setiap entri dalam kolom tersebut adalah NaN atau bukan, dan kemudian `.sum()` digunakan untuk menghitung total jumlah NaN dalam kolom tersebut.

Hasil dari perhitungan tersebut kemudian dicetak ke layar menggunakan perintah `print()` untuk masing-masing kolom, sehingga pengguna dapat melihat berapa banyak nilai NaN yang ada dalam setiap kolom tersebut.

Hasil dari perhitungan tersebut menunjukkan bahwa:
- Jumlah NaN pada kolom 'Bedroom2' adalah 0.
- Jumlah NaN pada kolom 'Bathroom' adalah 0.
- Jumlah NaN pada kolom 'Car' adalah 62.

In [56]:
# Menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Car'
data.dropna(subset=['Car'], inplace=True)


Kode di atas digunakan untuk menghapus baris (data) yang memiliki nilai yang hilang (NaN) dalam kolom 'Car' pada suatu dataframe atau dataset.

- `data` adalah nama dataframe yang digunakan.
- `.dropna()` adalah metode yang digunakan untuk menghapus baris-baris yang memiliki nilai yang hilang (NaN).
- `subset=['Car']` mengindikasikan bahwa penghapusan hanya akan dilakukan pada kolom 'Car'.
- `inplace=True` digunakan untuk mengubah dataframe `data` tanpa perlu membuat dataframe baru, sehingga perubahan tersebut akan dilakukan pada dataframe yang sama.

Dengan demikian, kode di atas akan menghasilkan dataframe yang telah dimodifikasi, di mana semua baris yang memiliki nilai yang hilang (NaN) pada kolom 'Car' telah dihapus.

In [59]:
# Menghitung jumlah nilai NaN (missing values) dalam kolom 'Car' dari DataFrame 'data'
nan_count_car = data['Car'].isna().sum()

# Menampilkan jumlah nilai NaN dalam kolom 'Car' menggunakan f-string
print(f"Jumlah nilai NaN dalam kolom Car: {nan_count_car}")


Jumlah nilai NaN dalam kolom Car: 0


Kode di atas digunakan untuk menghitung jumlah nilai NaN (missing values) dalam kolom 'Car' dari suatu DataFrame yang disebut 'data'. Langkah-langkah yang dilakukan dalam kode tersebut adalah sebagai berikut:

1. `nan_count_car = data['Car'].isna().sum()`: Di sini, kita mengambil kolom 'Car' dari DataFrame 'data' menggunakan `data['Car']`. Kemudian, kita menggunakan metode `.isna()` untuk menghasilkan Series yang berisi nilai True atau False, yang menunjukkan apakah setiap entri dalam kolom 'Car' adalah NaN atau tidak. Terakhir, kita menggunakan `.sum()` untuk menjumlahkan jumlah True (yang mewakili nilai NaN) dalam Series tersebut. Hasil perhitungan ini akan disimpan dalam variabel `nan_count_car`.

2. `print(f"Jumlah nilai NaN dalam kolom Car: {nan_count_car}")`: Pada langkah ini, kita menggunakan f-string untuk mencetak hasil perhitungan di atas dalam bentuk pesan. Pesan ini akan mencetak jumlah nilai NaN dalam kolom 'Car' ke layar.

Hasil dari kode ini adalah pesan "Jumlah nilai NaN dalam kolom Car: 0" yang menunjukkan bahwa dalam kolom 'Car' tidak ada nilai NaN, atau semua entri dalam kolom tersebut memiliki nilai yang valid.

In [61]:
# Mengambil bentuk (shape) dari data
data.shape


(13518, 21)

Kode di atas adalah perintah dalam bahasa pemrograman Python yang digunakan untuk mengambil bentuk (shape) dari sebuah variabel yang disebut "data." Bentuk data ini mengacu pada jumlah baris dan kolom yang ada dalam variabel tersebut. Dalam kasus ini, bentuk data "data" adalah (13518, 21), yang berarti variabel "data" memiliki 13.518 baris dan 21 kolom.

In [62]:
# Mengubah kolom 'Bedroom2' menjadi tipe data integer
data['Bedroom2'] = data['Bedroom2'].astype(int)

# Mengubah kolom 'Bathroom' menjadi tipe data integer
data['Bathroom'] = data['Bathroom'].astype(int)

# Mengubah kolom 'Car' menjadi tipe data integer
data['Car'] = data['Car'].astype(int)


Kode di atas digunakan untuk mengubah tipe data dari kolom-kolom tertentu dalam suatu dataframe yang disebut 'data'. Kolom-kolom yang diubah tipe datanya adalah 'Bedroom2', 'Bathroom', dan 'Car', yang semula mungkin memiliki tipe data yang berbeda menjadi tipe data integer (bilangan bulat).

Langkah-langkah yang diambil dalam kode tersebut adalah sebagai berikut:

1. `data['Bedroom2'] = data['Bedroom2'].astype(int)`: Kode ini mengambil kolom 'Bedroom2' dalam dataframe 'data' dan kemudian menggunakan metode `astype(int)` untuk mengubah tipe data dari kolom tersebut menjadi tipe data integer. Dengan demikian, kolom 'Bedroom2' akan hanya berisi bilangan bulat.

2. `data['Bathroom'] = data['Bathroom'].astype(int)`: Kode ini melakukan hal yang sama dengan langkah sebelumnya, tetapi pada kolom 'Bathroom'. Kolom 'Bathroom' yang semula mungkin memiliki tipe data lain akan diubah menjadi tipe data integer.

3. `data['Car'] = data['Car'].astype(int)`: Langkah ini juga mirip dengan langkah sebelumnya, tetapi diterapkan pada kolom 'Car'. Tipe data kolom 'Car' akan diubah menjadi integer.

Dengan mengubah tipe data kolom-kolom ini menjadi integer, Anda dapat melakukan operasi matematika atau analisis data yang melibatkan kolom-kolom ini dengan lebih mudah, karena tipe data integer sesuai untuk operasi aritmatika, perbandingan, dan analisis statistik.

In [63]:
# Pilih subset dari predictor (variabel independen) yang akan digunakan dalam analisis.
cols_to_use = ['Rooms', 'Distance', 'Landsize', 'BuildingArea', 'YearBuilt']

# Buat dataframe 'X' yang hanya berisi kolom-kolom yang terdapat dalam 'cols_to_use'.
X = data[cols_to_use]


Kode di atas digunakan untuk memilih subset dari variabel independen (prediktor) yang akan digunakan dalam analisis data. Variabel independen ini disimpan dalam sebuah DataFrame yang diberi nama 'X'. Berikut penjelasan lebih detailnya:

1. `cols_to_use` adalah sebuah daftar yang berisi nama-nama kolom yang akan digunakan sebagai prediktor dalam analisis. Dalam contoh ini, kita memilih lima kolom, yaitu 'Rooms', 'Distance', 'Landsize', 'BuildingArea', dan 'YearBuilt'.

2. Kemudian, kita membuat sebuah DataFrame baru yang disebut 'X' dengan menggunakan data dari DataFrame yang sudah ada (disebut 'data') dan hanya memilih kolom-kolom yang terdapat dalam daftar `cols_to_use`. Ini dilakukan dengan menggunakan tanda kurung siku dan nama kolom yang ingin dipilih.

Sebagai hasilnya, 'X' akan berisi hanya kolom-kolom yang tercantum dalam 'cols_to_use', dan ini akan menjadi subset dari data yang digunakan dalam analisis lebih lanjut. Ini berguna ketika Anda ingin fokus pada variabel-variabel tertentu dalam analisis statistik atau pembuatan model.

In [64]:
# Select target
y = data.Price


Kode di atas adalah potongan kode dalam bahasa pemrograman Python yang mengasumsikan ada variabel bernama `data` yang telah didefinisikan sebelumnya. Kode tersebut melakukan operasi berikut:

1. Mengambil nilai dari kolom atau atribut yang disebut "Price" dari variabel `data` dan menyimpannya dalam variabel baru yang dinamakan `y`. Ini mengasumsikan bahwa `data` adalah suatu struktur data yang berisi beberapa jenis data, seperti dataframe atau dictionary, yang memiliki atribut "Price" yang berisi data harga.

Hasil dari kode di atas adalah variabel `y` akan berisi data harga yang diambil dari `data`. Kode tersebut dapat digunakan untuk menganalisis atau memproses data harga lebih lanjut dalam konteks yang lebih luas.

In [65]:
from sklearn.model_selection import train_test_split

# Memisahkan data menjadi set pelatihan dan validasi
X_train, X_valid, y_train, y_valid = train_test_split(X, y, random_state=42, test_size=0.3)


Kode di atas menggunakan modul `train_test_split` dari pustaka scikit-learn (sklearn) untuk memisahkan dataset menjadi dua bagian: set pelatihan (training set) dan set validasi (validation set).

- `X` dan `y` adalah dua array atau matriks yang berisi fitur-fitur dan label-label dari dataset yang ingin dipisahkan.
- `random_state=42` digunakan untuk mengatur generator angka acak agar hasil pemisahan data menjadi set pelatihan dan validasi konsisten jika kode dieksekusi ulang dengan parameter yang sama.
- `test_size=0.3` mengindikasikan bahwa sekitar 30% dari data akan dialokasikan sebagai set validasi, sementara sisanya (70%) akan menjadi set pelatihan. Persentase pembagian ini dapat disesuaikan sesuai kebutuhan.

Hasil dari pemisahan ini adalah:
- `X_train` berisi data fitur dari set pelatihan.
- `X_valid` berisi data fitur dari set validasi.
- `y_train` berisi label-label yang sesuai dengan data fitur di set pelatihan.
- `y_valid` berisi label-label yang sesuai dengan data fitur di set validasi.

Dengan memisahkan dataset menjadi set pelatihan dan validasi, ini memungkinkan pengguna untuk melatih model machine learning pada set pelatihan dan mengukur kinerja model tersebut pada set validasi untuk menghindari overfitting dan mendapatkan perkiraan yang lebih baik tentang seberapa baik model akan berperform pada data yang belum pernah dilihat sebelumnya.

In [66]:
# Import library XGBRegressor dari XGBoost
from xgboost import XGBRegressor

# Membuat objek model XGBoost
my_model = XGBRegressor()

# Melatih model dengan data pelatihan X_train dan target y_train
my_model.fit(X_train, y_train)


XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=None, n_jobs=None,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas adalah contoh penggunaan library XGBoost dalam Python untuk membuat dan melatih model regresi menggunakan algoritma XGBoost. Berikut adalah penjelasan singkat dari setiap baris kode:

1. Import library XGBRegressor dari XGBoost:
   - Pada baris pertama, kita mengimpor kelas `XGBRegressor` dari library XGBoost. Ini adalah salah satu estimator yang disediakan oleh XGBoost untuk tugas regresi.

2. Membuat objek model XGBoost:
   - Pada baris kedua, kita membuat objek `my_model` yang akan digunakan untuk menampung model regresi XGBoost yang akan kita latih.

3. Melatih model dengan data pelatihan X_train dan target y_train:
   - Pada baris ketiga, kita menggunakan metode `fit()` pada objek `my_model` untuk melatih model XGBoost dengan menggunakan data pelatihan yang disimpan dalam `X_train` dan target yang disimpan dalam `y_train`. Proses pelatihan ini akan menghasilkan model regresi yang siap digunakan untuk melakukan prediksi.

Setelah kode ini dijalankan, `my_model` akan berisi model regresi XGBoost yang telah dilatih dengan data pelatihan yang sesuai. Model ini kemudian dapat digunakan untuk membuat prediksi pada data baru.

In [67]:
# Mengimpor modul mean_absolute_error dari sklearn.metrics
from sklearn.metrics import mean_absolute_error

# Melakukan prediksi menggunakan model (my_model) pada data validasi (X_valid)
predictions = my_model.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan nilai target aktual (y_valid)
# dan mencetak hasilnya
print("Mean Absolute Error: " + str(mean_absolute_error(predictions, y_valid)))


Mean Absolute Error: 241637.14228920118


Kode di atas mengimpor modul `mean_absolute_error` dari pustaka `sklearn.metrics`. Kemudian, menggunakan model yang disimpan dalam variabel `my_model`, melakukan prediksi pada data validasi yang tersimpan dalam variabel `X_valid`. Setelah prediksi selesai, kode menghitung Mean Absolute Error (MAE) antara prediksi yang dihasilkan oleh model dan nilai target aktual yang disimpan dalam variabel `y_valid`. Hasil MAE kemudian dicetak ke layar sebagai keluaran, sehingga kita dapat melihat seberapa besar kesalahan rata-rata prediksi model terhadap data validasi. Dalam kasus ini, MAE sebesar 241637.14228920118 menunjukkan rata-rata kesalahan absolut sekitar 241637.14228920118 antara prediksi model dan nilai target aktual pada data validasi.

In [68]:
# Membuat model XGBoost Regressor dengan 500 pohon keputusan (n_estimators=500)
my_model_0 = XGBRegressor(n_estimators=500)

# Melatih model menggunakan data pelatihan X_train dan label y_train
my_model_0.fit(X_train, y_train)


XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=None, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=500, n_jobs=None,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas menciptakan sebuah model XGBoost Regressor dengan 500 pohon keputusan (n_estimators=500) menggunakan library XGBoost. Model ini kemudian dilatih (fit) dengan data pelatihan yang disimpan dalam variabel X_train dan labelnya y_train. Model XGBoost Regressor ini memiliki berbagai parameter dan konfigurasi yang dapat diatur, meskipun dalam kode tersebut parameter-parameter tersebut tidak dijelaskan dengan rinci.

Berikut beberapa parameter utama yang dapat ditemukan dalam model XGBoost Regressor:

- n_estimators: Parameter ini mengatur jumlah pohon keputusan yang akan digunakan dalam model. Dalam kode di atas, n_estimators diatur ke 500, yang berarti model ini akan menggunakan 500 pohon keputusan dalam proses pembelajarannya.

- X_train: Ini adalah dataset yang digunakan untuk melatih model. Dataset ini berisi fitur-fitur yang digunakan untuk memprediksi target (y_train).

- y_train: Ini adalah label atau target yang digunakan dalam proses pelatihan. Model akan belajar untuk memprediksi nilai-nilai yang sesuai dengan data pelatihan ini.

Pada dasarnya, kode tersebut menciptakan dan melatih model regresi XGBoost dengan 500 pohon keputusan menggunakan data pelatihan X_train dan y_train. Model ini dapat digunakan untuk melakukan prediksi regresi pada data baru setelah dilatih.

In [70]:
# Melakukan prediksi menggunakan model 'my_model_0' pada data validasi 'X_valid'
predictions_0 = my_model_0.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan target sebenarnya 'y_valid'
mae = mean_absolute_error(predictions_0, y_valid)

# Menampilkan MAE ke layar
print("Mean Absolute Error: " + str(mae))


Mean Absolute Error: 254079.60437854414


Kode di atas merupakan contoh penggunaan sebuah model yang disebut 'my_model_0' untuk melakukan prediksi pada data validasi 'X_valid'. Setelah melakukan prediksi, kode tersebut menghitung Mean Absolute Error (MAE) antara prediksi yang dihasilkan oleh model dan nilai target sebenarnya yang ada dalam 'y_valid'. Hasil MAE kemudian ditampilkan di layar sebagai nilai numerik.

MAE merupakan metrik evaluasi yang digunakan untuk mengukur seberapa besar kesalahan rata-rata antara prediksi model dan nilai target sebenarnya. Dalam konteks ini, MAE bernilai sekitar 254,079.60, yang menunjukkan rata-rata kesalahan prediksi model 'my_model_0' terhadap data validasi 'X_valid' sekitar 254,079.60. Semakin kecil nilai MAE, semakin baik kualitas model dalam melakukan prediksi, karena nilai kesalahan yang lebih rendah menunjukkan bahwa prediksi lebih mendekati nilai sebenarnya.

In [71]:
from sklearn.model_selection import GridSearchCV

# Tentukan model XGBoost
model_grid = XGBRegressor()

# Tentukan parameter grid
param_grid = {
    'n_estimators': [150, 200, 250, 300, 1000]  # Ganti dengan rentang nilai yang ingin Anda coba
}

# Inisialisasi GridSearchCV
grid_search = GridSearchCV(estimator=model_grid, param_grid=param_grid, cv=5, scoring='neg_mean_absolute_error')

# Lakukan fitting menggunakan data training
grid_search.fit(X_train, y_train)

# Dapatkan parameter terbaik
best_n_estimators = grid_search.best_params_['n_estimators']

print("Nilai n_estimators terbaik:", best_n_estimators)
# Tentukan range nilai n_estimators yang akan diuji
n_estimators_list = [100, 150, 200, 250, 300]

# Inisialisasi sebuah list untuk menyimpan hasil MAE
mae_results = []

# Loop melalui setiap nilai n_estimators
for n_estimators in n_estimators_list:
    # Inisialisasi model dengan n_estimators tertentu
    model_grid = XGBRegressor(n_estimators=n_estimators)

    # Fitting model menggunakan data training
    model_grid.fit(X_train, y_train)

    # Lakukan prediksi pada data validasi
    predictions = model_grid.predict(X_valid)

    # Hitung Mean Absolute Error
    mae = mean_absolute_error(predictions, y_valid)

    # Tampilkan nilai n_estimators dan MAE
    print(f"n_estimators: {n_estimators}, Mean Absolute Error: {mae}")

    # Tambahkan nilai MAE ke dalam list
    mae_results.append(mae)

Nilai n_estimators terbaik: 150
n_estimators: 100, Mean Absolute Error: 241637.14228920118
n_estimators: 150, Mean Absolute Error: 243228.3698976054
n_estimators: 200, Mean Absolute Error: 245296.04482749323
n_estimators: 250, Mean Absolute Error: 247178.43902937006
n_estimators: 300, Mean Absolute Error: 249329.28532189966


Kode di atas merupakan contoh penggunaan GridSearchCV dari library scikit-learn (sklearn) untuk melakukan pencarian parameter terbaik pada model XGBoost. GridSearchCV adalah metode untuk melakukan pencarian hiperparameter terbaik dari suatu model dengan mencoba berbagai kombinasi nilai parameter yang telah ditentukan sebelumnya.

1. Import modul yang dibutuhkan:
   ```python
   from sklearn.model_selection import GridSearchCV
   ```

2. Tentukan model XGBoost:
   ```python
   model_grid = XGBRegressor()
   ```

3. Tentukan parameter grid yang akan diuji dalam pencarian hiperparameter. Dalam kode ini, hanya satu parameter yang diuji, yaitu 'n_estimators', dengan beberapa nilai yang mungkin.
   ```python
   param_grid = {
       'n_estimators': [150, 200, 250, 300, 1000]
   }
   ```

4. Inisialisasi objek GridSearchCV dengan model, parameter grid, metode validasi silang (cross-validation), dan metrik evaluasi (scoring) yang ingin digunakan (dalam kasus ini, 'neg_mean_absolute_error' digunakan sebagai metrik untuk evaluasi).
   ```python
   grid_search = GridSearchCV(estimator=model_grid, param_grid=param_grid, cv=5, scoring='neg_mean_absolute_error')
   ```

5. Lakukan fitting (pelatihan) menggunakan data training (X_train dan y_train) untuk mencari parameter terbaik.
   ```python
   grid_search.fit(X_train, y_train)
   ```

6. Dapatkan parameter terbaik dari hasil pencarian GridSearchCV.
   ```python
   best_n_estimators = grid_search.best_params_['n_estimators']
   ```

7. Cetak nilai 'n_estimators' terbaik.
   ```python
   print("Nilai n_estimators terbaik:", best_n_estimators)
   ```

8. Selanjutnya, kode melakukan pencarian parameter 'n_estimators' dengan nilai tertentu di luar GridSearchCV. Ini digunakan untuk membandingkan kinerja model XGBoost dengan berbagai nilai 'n_estimators'.
   ```python
   n_estimators_list = [100, 150, 200, 250, 300]
   ```

9. Inisialisasi list 'mae_results' untuk menyimpan hasil Mean Absolute Error (MAE) dari setiap percobaan nilai 'n_estimators'.
   ```python
   mae_results = []
   ```

10. Dilakukan loop untuk masing-masing nilai 'n_estimators'. Dalam setiap iterasi, model XGBoost diinisialisasi dengan 'n_estimators' tertentu, kemudian dilakukan fitting menggunakan data training.
   ```python
   for n_estimators in n_estimators_list:
       model_grid = XGBRegressor(n_estimators=n_estimators)
       model_grid.fit(X_train, y_train)
   ```

11. Setelah model difitting, dilakukan prediksi pada data validasi (X_valid) dan dihitung MAE dari prediksi tersebut.
   ```python
   predictions = model_grid.predict(X_valid)
   mae = mean_absolute_error(predictions, y_valid)
   ```

12. Hasil 'n_estimators' dan MAE dari setiap iterasi dicetak.
   ```python
   print(f"n_estimators: {n_estimators}, Mean Absolute Error: {mae}")
   ```

13. Nilai MAE dari setiap percobaan disimpan dalam list 'mae_results' untuk analisis lebih lanjut.
   ```python
   mae_results.append(mae)
   ```

In [72]:
# Membuat model XGBoost Regressor dengan 100 pohon keputusan (n_estimators=100)
my_model_2 = XGBRegressor(n_estimators=100)

# Melatih model menggunakan data pelatihan (X_train, y_train) dengan early stopping
# untuk menghentikan pelatihan jika tidak ada perbaikan dalam 5 iterasi
# dan menggunakan data validasi (X_valid, y_valid) untuk evaluasi
my_model_2.fit(X_train, y_train, early_stopping_rounds=5, eval_set=[(X_valid, y_valid)], verbose=False)

# Melakukan prediksi pada data validasi menggunakan model yang telah dilatih
predictions_2 = my_model_2.predict(X_valid)

# Menghitung Mean Absolute Error (MAE) antara prediksi dan nilai sebenarnya pada data validasi
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_2, y_valid)))


/usr/local/lib/python3.10/dist-packages/xgboost/sklearn.py:885: UserWarning: `early_stopping_rounds` in `fit` method is deprecated for better compatibility with scikit-learn, use `early_stopping_rounds` in constructor or`set_params` instead.
  warnings.warn(


Mean Absolute Error: 246166.56518602843


Kode di atas adalah contoh penggunaan model XGBoost Regressor untuk melakukan prediksi pada data validasi dan mengukur kinerja model menggunakan metrik Mean Absolute Error (MAE). Berikut penjelasan dari setiap langkah dalam kode tersebut:

1. Membuat model XGBoost Regressor dengan 100 pohon keputusan (n_estimators=100). Model ini akan digunakan untuk melakukan prediksi pada data.

2. Melatih model menggunakan data pelatihan (X_train, y_train) dengan parameter tambahan `early_stopping_rounds=5`. Ini mengindikasikan bahwa pelatihan model akan berhenti jika tidak ada perbaikan dalam nilai evaluasi pada data validasi selama 5 iterasi. Data validasi digunakan untuk mengevaluasi kinerja model selama pelatihan.

3. Melakukan prediksi pada data validasi (X_valid) menggunakan model yang telah dilatih. Hasil prediksi disimpan dalam variabel `predictions_2`.

4. Menghitung Mean Absolute Error (MAE) antara hasil prediksi (predictions_2) dan nilai sebenarnya (y_valid) pada data validasi. MAE adalah metrik yang mengukur sejauh mana prediksi model dari nilai sebenarnya dalam satuan absolut. Hasil MAE dicetak ke layar sebagai keluaran.

Selain itu, pesan peringatan dalam output yang mengatakan "early_stopping_rounds" dalam metode "fit" sudah tidak direkomendasikan dan harus digunakan dalam konstruktor atau "set_params" adalah peringatan yang menginformasikan bahwa penggunaan parameter `early_stopping_rounds` dalam metode `fit` di XGBoost sudah tidak disarankan dan seharusnya digunakan dalam konstruktor atau `set_params` untuk lebih baik kompatibilitas dengan scikit-learn.

In [73]:
# Import library XGBRegressor untuk mengembangkan model XGBoost
my_model_1 = XGBRegressor(n_estimators=100)

# Melatih model menggunakan data pelatihan
my_model_1.fit(X_train, y_train)

# Membuat prediksi menggunakan model yang telah dilatih pada data validasi
predictions_1 = my_model_1.predict(X_valid)

# Menghitung dan mencetak Mean Absolute Error (MAE) dari prediksi model
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_1, y_valid)))


Mean Absolute Error: 241637.14228920118


Kode di atas adalah contoh penggunaan library XGBoost untuk mengembangkan model regresi dengan XGBRegressor. Berikut adalah penjelasan langkah-langkahnya:

1. Pertama-tama, library XGBRegressor diimpor ke dalam program. Ini adalah library yang digunakan untuk mengembangkan model regresi dengan algoritma XGBoost.

2. Model regresi XGBoost didefinisikan dengan menggunakan `XGBRegressor(n_estimators=100)`. `n_estimators` adalah parameter yang menentukan jumlah pohon keputusan yang akan digunakan dalam model. Dalam kasus ini, model akan terdiri dari 100 pohon keputusan.

3. Selanjutnya, model tersebut dilatih dengan menggunakan data pelatihan. Data pelatihan diberikan dalam bentuk `X_train` (fitur) dan `y_train` (label/target).

4. Setelah model dilatih, digunakan untuk membuat prediksi pada data validasi (`X_valid`).

5. Mean Absolute Error (MAE) dihitung dengan membandingkan prediksi yang dihasilkan oleh model (`predictions_1`) dengan data validasi (`y_valid`). MAE adalah metrik evaluasi yang mengukur seberapa dekat prediksi model dengan nilai sebenarnya. Semakin rendah MAE, semakin baik kualitas model regresi.

6. Hasil MAE dicetak ke layar untuk memberikan informasi tentang sejauh mana model dapat memprediksi data validasi dengan akurasi.

Hasil keluaran dari kode di atas adalah nilai MAE yang mengindikasikan tingkat kesalahan prediksi model terhadap data validasi. Dalam contoh ini, MAE memiliki nilai sekitar 241,637.14, yang menunjukkan sejauh mana model dapat memprediksi data validasi dengan rata-rata kesalahan sekitar 241,637.14 unit. Semakin rendah nilai MAE, semakin baik kinerja model dalam memprediksi data.

In [74]:
# Membuat model XGBoost Regressor dengan 100 pohon keputusan (n_estimators=100).
# Model ini akan digunakan untuk memprediksi variabel target berdasarkan fitur-fitur dalam dataset.
my_model_3 = XGBRegressor(n_estimators=100, learning_rate=0.05, n_jobs=6)

# Melatih model dengan menggunakan data pelatihan (X_train) dan target (y_train).
my_model_3.fit(X_train, y_train)



XGBRegressor(base_score=None, booster=None, callbacks=None,
             colsample_bylevel=None, colsample_bynode=None,
             colsample_bytree=None, device=None, early_stopping_rounds=None,
             enable_categorical=False, eval_metric=None, feature_types=None,
             gamma=None, grow_policy=None, importance_type=None,
             interaction_constraints=None, learning_rate=0.05, max_bin=None,
             max_cat_threshold=None, max_cat_to_onehot=None,
             max_delta_step=None, max_depth=None, max_leaves=None,
             min_child_weight=None, missing=nan, monotone_constraints=None,
             multi_strategy=None, n_estimators=100, n_jobs=6,
             num_parallel_tree=None, random_state=None, ...)

Kode di atas menciptakan sebuah model regresi menggunakan algoritma XGBoost dengan 100 pohon keputusan (n_estimators=100) yang akan digunakan untuk memprediksi variabel target berdasarkan fitur-fitur dalam dataset. Model ini ditentukan dengan konfigurasi berikut:

1. `XGBRegressor(n_estimators=100, learning_rate=0.05, n_jobs=6)`: Model XGBoost Regressor dibuat dengan beberapa parameter konfigurasi, termasuk:
   - `n_estimators=100`: Model ini akan terdiri dari 100 pohon keputusan yang akan digunakan dalam proses prediksi.
   - `learning_rate=0.05`: Tingkat pembelajaran yang digunakan dalam algoritma XGBoost. Ini mengontrol sejauh mana setiap pohon keputusan memperbarui prediksi model sebelumnya.
   - `n_jobs=6`: Jumlah pekerjaan paralel yang akan digunakan dalam pelatihan model. Dalam hal ini, 6 pekerjaan akan digunakan untuk melatih model.

2. `my_model_3.fit(X_train, y_train)`: Model XGBoost ini dilatih menggunakan data pelatihan yang diberikan dalam `X_train` dan target yang diberikan dalam `y_train`. Proses pelatihan ini akan menghasilkan model yang dapat digunakan untuk membuat prediksi berdasarkan data yang tidak terlihat sebelumnya.

Selain parameter-parameter yang disebutkan di atas, terdapat juga parameter lain yang bisa dikustomisasi dalam model XGBoost, seperti `max_depth`, `gamma`, `colsample_bytree`, dan banyak lagi. Model ini diarahkan untuk melakukan regresi, yaitu memprediksi nilai berkelanjutan, dan akan mencoba memahami pola dalam data pelatihan untuk melakukan prediksi yang akurat terhadap target yang diberikan.

In [75]:
# predictions_3 adalah prediksi yang dihasilkan oleh model (my_model_3) terhadap data validasi (X_valid).
predictions_3 = my_model_3.predict(X_valid)

# mean_absolute_error adalah fungsi untuk menghitung Mean Absolute Error (MAE) antara prediksi (predictions_3) dan nilai sebenarnya (y_valid).
# Hasil MAE ini mengukur sejauh mana prediksi model berbeda dari nilai sebenarnya dalam skala absolut.
# Hasil MAE kemudian dicetak sebagai teks.
print("Mean Absolute Error: " + str(mean_absolute_error(predictions_3, y_valid)))


Mean Absolute Error: 254714.15406958826


Kode di atas digunakan untuk menghitung Mean Absolute Error (MAE) antara prediksi yang dihasilkan oleh model (my_model_3) terhadap data validasi (X_valid) dan nilai sebenarnya (y_valid). Hasil MAE ini mengukur sejauh mana prediksi model berbeda dari nilai sebenarnya dalam skala absolut. Hasil MAE kemudian dicetak sebagai teks, yang dalam contoh ini adalah sekitar 254714.15406958826.